crates/configuration/src/version1.rs

```diff
@@ -181,12 +165,17 @@ async fn configure_stored_procedures(
 ) -> Result<StoredProcedures, Error> {
     match config_options {
         Some(config_options) => {
-            let stored_procedures_row =
-                select_first_row(mssql_pool, STORED_PROCS_CONFIGURATION_QUERY).await;
-            let introspected_stored_procedures: Vec<introspection::IntrospectStoredProcedure> =
-                serde_json::from_str(stored_procedures_row.get(0).unwrap())
-                    .map_err(Error::StoredProcedureIntrospectionError)?;
-            let new_stored_procedures = get_stored_procedures(introspected_stored_procedures);
+            let mut connection = mssql_pool
+                .get()
+                .await.map_err(Error::GetConnectionFromPool)?;
+            // Let's do some stored procedures introspection
+            let mut stored_procs_query = SQL::new();
+            RawSql::RawText(STORED_PROCS_CONFIGURATION_QUERY.to_string()).to_sql(&mut stored_procs_query);
+            let mut stored_procs_rows = Vec::new();
```

Comment:
I don't feel like I understand what's going on here with this Vec and `execute_query`. From my understanding there is only one row returned by the introspection query... but the way this is implemented, each row's string is just blindly concatenated into the same array of chars here (`stored_procs_row` is `Vec<u8>`). That seems... dubious.

My quick reading of the tiberius API we're using is that we should be taking a QueryStream and calling `into_first_result`, which gives a Vec of Row. Then we should extract our JSON from the one column in the one row we find there and error if we get more than one row because why are our JSON queries returning multiple rows...

The fact that our logic reads multiple rows and munges all data from the first column in each row into a single string is scary. I can't find anything my quick check of the SQL Server docs or AI queries that indicate that a JSON string would be split across multiple rows which is what the execute_query code seems to be saying will happen.

Reply:
I'll create a ticket to do this in a follow up. Good call out.

Reply:
Created ticket here: https://linear.app/hasura/issue/ENT-152/document-and-refactor-execute-query-in-ndc-sqlserver